CMakeLists.txt

```diff
@@ -48,19 +48,134 @@ if(POLICY CMP0135)
+    )
+  else()
+    if(NOT DEFINED BUILDENV_NAME)
+      if(DEFINED ENV{BUILDENV_NAME})
+        set(BUILDENV_NAME "$ENV{BUILDENV_NAME}")
+      else()
+        message(
+          FATAL_ERROR
+          "Neither BUILDENV_URL nor BUILDENV_NAME specified!\nDid you run `${CMAKE_SOURCE_DIR}/tools/windows_buildenv.bat`?"
+        )
+      endif()
+    endif()
+  endif()
+
+  message(STATUS "BUILDENV_NAME is: ${BUILDENV_NAME}")
+
+  if(
+    EXISTS "${BUILDENV_BASEPATH}/${BUILDENV_NAME}"
+    AND IS_DIRECTORY "${BUILDENV_BASEPATH}/${BUILDENV_NAME}"
+  )
+    message(
+      STATUS
+      "BUILDENV directory ${BUILDENV_BASEPATH}/${BUILDENV_NAME} exists!"
+    )
+  else()
+    if(NOT DEFINED BUILDENV_URL)
+      message(
+        FATAL_ERROR
+        "BUILDENV_URL not specified!\nDid you run `${CMAKE_SOURCE_DIR}/tools/windows_buildenv.bat`?"
+      )
+    endif()
+    if(NOT DEFINED BUILDENV_SHA256)
+      if(DEFINED ENV{BUILDENV_SHA256})
+        set(BUILDENV_SHA256 "$ENV{BUILDENV_SHA256}")
+      else()
+        message(STATUS "BUILDENV_SHA256 not specified")
+      endif()
+    endif()
+
+    if(NOT EXISTS "${BUILDENV_BASEPATH}/${BUILDENV_NAME}.zip")
+      message(
+        STATUS
+        "Downloading file ${BUILDENV_URL} to ${BUILDENV_BASEPATH} ..."
+      )
+      file(
+        DOWNLOAD ${BUILDENV_URL} "${BUILDENV_BASEPATH}/${BUILDENV_NAME}.zip"
+        SHOW_PROGRESS
+        TLS_VERIFY ON
+      )
+    endif()
+
+    if(NOT ${BUILDENV_SHA256} STREQUAL "")
+      message(
+        STATUS
+        "Verify SHA256 of downloaded file ${BUILDENV_BASEPATH}/${BUILDENV_NAME}.zip ..."
+      )
+      file(SHA256 "${BUILDENV_BASEPATH}/${BUILDENV_NAME}.zip" actual_sha256)
+
+      if(NOT actual_sha256 STREQUAL ${BUILDENV_SHA256})
+        message(
+          FATAL_ERROR
+          "SHA256 checksum mismatch:\nexpected: ${BUILDENV_SHA256}\n     got: ${actual_sha256}"
+        )
+      else()
+        message(STATUS "SHA256 ${BUILDENV_SHA256} is correct!")
+      endif()
+    endif()
+
+    message(
+      STATUS
+      "Unpacking file ${BUILDENV_BASEPATH}/${BUILDENV_NAME}.zip ..."
+    )
+    execute_process(
+      COMMAND ${CMAKE_COMMAND} -E tar xzf "${BUILDENV_NAME}.zip"
+      WORKING_DIRECTORY ${BUILDENV_BASEPATH}
+      RESULT_VARIABLE unpack_result
+    )
+    if(NOT unpack_result EQUAL 0)
+      message(FATAL_ERROR "Failed to unpack ${BUILDENV_NAME}.zip")
+    else()
+      file(REMOVE "${BUILDENV_BASEPATH}/${BUILDENV_NAME}.zip")
+      message(STATUS "Deleted ${BUILDENV_NAME}.zip after successful extraction")
+    endif()
+  endif()
+endif()
+
 # Use this function to throw an error because the build environment is not set
 # up correctly.
 function(fatal_error_missing_env)
   if(WIN32)
     if(CMAKE_BUILD_TYPE MATCHES "Debug")
       message(
         FATAL_ERROR
-        "Did you download the Mixxx build environment using `${CMAKE_SOURCE_DIR}/tools/windows_buildenv.bat`?"
+        "Did you run `${CMAKE_SOURCE_DIR}/tools/windows_buildenv.bat`?"
```

Comment:
The information that a GB download is started by this is missing. That can be an issue if you are out of disk space or using a slow or expensive internet connection. 
Can we add this info to the windows_buildenv.bat 

Some think like this: 

```
echo Note: 
echo This will download the Mixxx build environment which is > 1 GB.
echo Make sure you have enough disk space and a fast connection with an unlimited plan.
echo.
choice /C YN /N /M "Press Y to continue or N to cancel: "
if errorlevel 2 (
    exit /b
)
```

Reply:
The batch file does not download anymore, it just writes a settings file and sets some environment variables.
The main goal of this PR is to get rid of application logic in this nasty batch file. The last thing we should do is to add new functionality to this batch file.

Reply:
This happens now "delayed". I think a typical run is like that: 
* Use cmake without the environment set (no download)  
* Read this prompt 
* Run the batch  
* Run cmake again 
* Being surprised that a GB download is started.

Before the situation was not ideal, but sightly better because only the dedicated batch call did the download, which is documented in the wiki. 

I have stumbled here because you have removed the info that a download happens. It still happens after running the script, but during the next cmake call. This can happen more or less nice by the IDE. 

Do you have an idea  how to improve the situation?

Reply:
I will rephrase the messages, to provide the removed information in a way that matchs to the new flow.

Reply:
I changed the messages and now the information about the download is back.